### DISCUSSION TITLE
Cannot run speaker verification example because of XA must be a 2-dimensional array

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [pyannote/pyannote-audio#1137] [General]
Hi,

First of all thanks again for this beautiful library!

I followed [the speaker verification example](https://huggingface.co/pyannote/embedding) but got : 
`ValueError: XA must be a 2-dimensional array.`.

```
from pyannote.audio import Model
model = Model.from_pretrained("pyannote/embedding", 
                              use_auth_token="ACCESS_TOKEN_GOES_HERE")

from pyannote.audio import Inference
inference = Inference(model, window="whole")
embedding1 = inference("speaker1.wav")
embedding2 = inference("speaker2.wav")
# `embeddingX` is (1 x D) numpy array extracted from the file as a whole.

from scipy.spatial.distance import cdist
distance = cdist(embedding1, embedding2, metric="cosine")[0,0]
# `distance` is a `float` describing how dissimilar speakers 1 and 2 are.

```

 So I followed [this answer from SO](https://stackoverflow.com/a/43873680/17524305) and applied the same `reshape()` on all embeddingX but depending on the reshape I get different values.

a. For example reshape(-1,1) will yield a distance of '0' between wav files from the same speaker, whereas it will give '2' between wav files from different speakers.

b. On the contrary reshape(1,-1) will give '0.42' for wav files from the same speaker and '0.91' for wav files from different speakers.

Consequently I am really not sure if I am doing things correctly because :
1. I am not able to make the huggingface example running
2. From a. I get '0' for similar voice but '2' for different voices (I would expect a normalized value, so '1') and `cdist` outputs a matrix.
3. From b. I get floats as expected but '0.42' where '0' was expected and '0.91' where '1' was expected.

So could you help me solving this problem ?

Thank you

**Comment by olliethedev** [comment]
I am having the same issue! Was anyone successful with the embeddings example?

**Reply by Ca-ressemble-a-du-fake** [reply]
I switched to another library (NeMo from nvidia).

**Reply by olliethedev** [reply]
So I actually ended up getting it to work with the following approach:
```python
import numpy as np
from scipy.spatial.distance import cdist
from pyannote.audio import Inference
inference = Inference(embeddingmodel, window="whole", device="cuda")
...
embedding1 = inference(file1.wav)
embedding2 = inference(file2.wav)
distance = cdist(np.reshape(embedding1, (1, -1)), np.reshape(embedding2, (1, -1)), metric="cosine")[0,0]
# closer the distance value is to 0 the more likely the speaker is the same person.
```
